scripts/mbedtls_framework/psa_information.py

```diff
@@ -123,38 +139,6 @@ def generate_deps_from_description(
 
     return dep_list
 
-# A temporary hack: at the time of writing, not all dependency symbols
-# are implemented yet. Skip test cases for which the dependency symbols are
-# not available. Once all dependency symbols are available, this hack must
-# be removed so that a bug in the dependency symbols properly leads to a test
-# failure.
-def read_implemented_dependencies(filename: str) -> FrozenSet[str]:
-    return frozenset(symbol
-                     for line in open(filename)
-                     for symbol in re.findall(r'\bPSA_WANT_\w+\b', line))
-_implemented_dependencies = None #type: Optional[FrozenSet[str]] #pylint: disable=invalid-name
-def hack_dependencies_not_implemented(dependencies: List[str]) -> None:
-    """
-    Hack dependencies to skip test cases for which at least one dependency
-    symbol is not available yet.
-    """
-    global _implemented_dependencies #pylint: disable=global-statement,invalid-name
-    if _implemented_dependencies is None:
-        # Temporary, while Mbed TLS does not just rely on the TF-PSA-Crypto
-        # build system to build its crypto library. When it does, the first
-        # case can just be removed.
-        if os.path.isdir('tf-psa-crypto'):
-            _implemented_dependencies = \
-                read_implemented_dependencies('tf-psa-crypto/include/psa/crypto_config.h')
-        else:
-            _implemented_dependencies = \
-                read_implemented_dependencies('include/psa/crypto_config.h')
-
-    if not all((dep.lstrip('!') in _implemented_dependencies or
-                not dep.lstrip('!').startswith('PSA_WANT'))
-               for dep in dependencies):
-        dependencies.append('DEPENDENCY_NOT_IMPLEMENTED_YET')
-
 def tweak_key_pair_dependency(dep: str, usage: str):
     """
```

Comment:
`psa_information.tweak_key_pair_dependency`, and thus `psa_test_case.TestCase.infer_dependencies`, silently suppresses dependencies on key pairs if a usage is specified and is not `BASIC` or `GENERATE` (for example `IMPORT` or `DERIVE`). Furthermore, if no usage is specified with `set_key_pair_usage()` and a key pair type is present, then `infer_dependencies` ends up generating the deprecated dependency symbol `PSA_WANT_KEY_TYPE_xxx_KEY_PAIR`.

Right now, this ends up working, because we always call either `set_key_pair_usage('GENERATE')` or `set_key_pair_usage('BASIC')`. This could be considered in scope since `set_key_pair_usage()` is a new method and it isn't working correctly. This could also be considered out of scope since it doesn't break anything and the goal of this pull request is to forward-port some improvements, not to fix every problem we can find. I propose to consider it out-of-scope here, and handle it in the follow-up that handles negative test cases, for which the current behavior does introduce a bug.

Reply:
Agreed, let's consider it out of scope. (I am not reviewing, just commenting.)

Reply:
I agree, we should consider it out of scope. (Just commenting, not reviewing.)